### ISSUE TITLE
Dependency Versioning

### ISSUE THREAD

**Issue by roshanr10** [vanna-ai/vanna#236]
Currently Vanna does not require specific version(s) of Python modules, opening up to potential dep mismatch over time – would love to add that in pyproject.toml, are there specific versions the development team tests against?

**Comment by relic-yuexi** [comment]
original need python > 3.9, because some type check need python>3.9. For example, list is not allowed for python 3.8, it must be List.

**Comment by roshanr10** [comment]
makes sense for Python version, my point is for other dependencies, i.e. if snowflake driver has a breaking change, or chromadb changes, or OpenAI, Vanna may not be able to integrate anymore, usually packages pin other dependencies to avoid this issue, see Vanna reps vs FastAPI deps below for ref: 

usually this tends to manifest as hard to track "it works on my system" issues b/c while you may have already the dependencies, the next install on a different machine may even pull a completely different major version with incompatible method signatures; this is a big concern with respect to production use for me

Vanna:
```
[project.optional-dependencies]
postgres = ["psycopg2-binary", "db-dtypes"]
bigquery = ["google-cloud-bigquery"]
snowflake = ["snowflake-connector-python"]
duckdb = ["duckdb"]
all = ["psycopg2-binary", "db-dtypes", "google-cloud-bigquery", "snowflake-connector-python","duckdb"]
test = ["tox"]
chromadb = ["chromadb"]
openai = ["openai"]
mistralai = ["mistralai"]
gemini = ["google-generativeai"]
marqo = ["marqo"]
```

FastAPI
```
[project.optional-dependencies]
all = [
    "httpx >=0.23.0",
    "jinja2 >=2.11.2",
    "python-multipart >=0.0.7",
    "itsdangerous >=1.1.0",
    "pyyaml >=5.3.1",
    "ujson >=4.0.1,!=4.0.2,!=4.1.0,!=4.2.0,!=4.3.0,!=5.0.0,!=5.1.0",
    "orjson >=3.2.1",
    "email_validator >=2.0.0",
    "uvicorn[standard] >=0.12.0",
    "pydantic-settings >=2.0.0",
    "pydantic-extra-types >=2.0.0",
]
```

**Comment by zainhoda** [comment]
@roshanr10 yeah, one thing we have to pin down is the actual version dependencies. One thing we don't want to do is force upgrades when they aren't necessary. The only true hard version dependency that we have is openai >= 1.0.0 

Since we aren't pinning on our end, what you could do is `pip freeze` the versions of these packages that you currently have installed and add them as requirements directly in your `requirements.txt` for now
